components/select/select.ts

```diff
@@ -87,11 +87,21 @@ export class SuiSelect<T, U> extends SuiSelectBase<T, U> {
         if (value != null) {
             if (this.options.length > 0) {
                 // If the options have already been loaded, we can immediately match the ngModel value to an option.
-                this.selectedOption = this.options.find(o => value == this.valueGetter(o));
+                this.selectedOption = this.findOption(this.options, value);
             }
             if (!this.selectedOption) {
-                // Otherwise, cache the written value for when options are set.
-                this._writtenOption = value;
+                if (this.searchService.selectedLookup) {
```

Comment:
Yes, maybe it is more correct that a search service is responsible for understanding if a `optionsLookup` function can also be a `selectedLookup` or not. I can do that if you agree. I have already moved the initialization lookup call to `writeValue` and pushed the commit.